sdk/helper/keysutil/policy.go

```diff
@@ -1169,7 +1190,12 @@ func (p *Policy) Sign(ver int, context, input []byte, hashAlgorithm HashType, si
 
 		switch sigAlgorithm {
 		case "pss":
-			sig, err = rsa.SignPSS(rand.Reader, key, algo, input, nil)
+			// At the time of writing, this constant has the lowest sentinel value:
+			// https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/crypto/rsa/pss.go;l=247
+			if saltLength < rsa.PSSSaltLengthEqualsHash {
+				return nil, errutil.UserError{Err: fmt.Sprintf("requested salt length %d is invalid", saltLength)}
+			}
```

Comment:
We would also need an upper bounds test here ideally, something possibly along these lines, again untested code, to give an idea, to replicate the test done in the lower layers of pss.go https://github.com/golang/go/blob/master/src/crypto/rsa/pss.go#L49-L53 but sadly happen too late.

```go
if err := verifyMaxPssSaltLength(key, algo, saltLength); err != nil {
	return nil, errutil.UserError{Err: fmt.Sprintf("requested salt length %d is too large", saltLength)}
}

func verifyMaxPssSaltLength(key *rsa.PrivateKey, hash crypto.Hash, saltLength int) error {
	emBits := key.N.BitLen() - 1
	hLen := hash.Size()
	emLen := (emBits + 7) / 8

	if emLen < hLen+saltLength+2 {
		return errors.New("pss salt value is too long")
	}

	return nil
}
```

Reply:
Yeah, I was conflicted about this. `crypto/rsa` doesn't seem to expose a public function to compute the upper bound, and I didn't wanna duplicate the logic (not that it's likely to change unless there is a bug). Let me see what I can do.

Reply:
Agreed, it isn't the best but unfortunately their seems to be an unchecked buffer allocation with the salt value passed in at https://github.com/golang/go/blob/master/src/crypto/rsa/pss.go#L302 so we definitely need to have an upper bounds check of some sort here.